docs: Adding a new language documentation (#4006)
mealie-recipes/mealie: docs/docs/contributors/developers-guide/adding-language.md

# How to add a language

> 1. A language must be added into [Crowdin](https://crowdin.com/project/mealie). This is typically done by hay-kot, and should be asked directly onto the [Discord server](https://discord.gg/QuStdQGSGK) in the #translation channel.

Reviewer: ```suggestion
1. The language must be added to [Crowdin](https://crowdin.com/project/mealie). Create an issue on GitHub requesting the language, or you may request it on the [official Mealie Discord server](https://discord.gg/QuStdQGSGK) in the `#translation` channel
```

> 2. After the language is added, an automatic PR is created and accepted. This can take some hours.

Reviewer: ```suggestion
2. After the language is added to Crowdin, a PR is automatically created, which will then be merged by one of the maintainers. It may take several hours for the PR to be created in GitHub.
```

> 3. Some files needs to be adapted, check the Changes required section here under

Reviewer: ```suggestion
3. Once the PR is merged, some files need to be manually updated in a PR. See the "Changes required" below.
```

## Changes required

> Another commit is required so that the language is taken into account.

Reviewer: ```suggestion
Another pull request is required so that the language is fully implemented.
```

> You can find an example commit [here](https://github.com/mealie-recipes/mealie/pull/4004/commits/0e65dd72d9ea6e35ab7caaf8513393cee7e0e9e9)

Reviewer: ```suggestion
You can find an example commit [here](https://github.com/mealie-recipes/mealie/pull/4004/commits/0e65dd72d9ea6e35ab7caaf8513393cee7e0e9e9).
```

**gen_ts_locales.py (dev/code-generation/gen_ts_locales.py)**

> A line has to be added with this format :

Reviewer: ```suggestion
A line has to be added with this format:
```

```
    "[Locale]": LocaleData(name="[Localized name] ([English name])"),
```

> where :

Reviewer: ```suggestion
Where:
```

> - **Locale** is the ICU locale with the language followed by an underscore and the country in uppercase : fr-FR, en-US, es-ES

Reviewer: ```suggestion
- **Locale** is the ICU locale with the language followed by an underscore and the country in uppercase (e.g. fr-FR, en-US, es-ES, ...)
```

- **Localized name** is the name of the language in the language itself : Français, English, Español

> - **English name** is the name of the language translated into english : French, Spanish, ..

Reviewer: ```suggestion
- **English name** is the name of the language translated into English (e.g. French, Spanish, ...)
```

**available-locales.ts (frontend/composables/use-locales/available-locales.ts)**

> Add a block specifying the language available for the frontend :

Reviewer: ```suggestion
Add a block specifying the language available for the frontend:
```

```
    {
        name: "[Localized name] ([English name])",
        value: "[Locale]",
        progress: 100,
        dir: "ltr",
    },
```

**Date time format (frontend/lang/dateTimeFormats/[locale].json)**

> A file needs to be added for the date time formats

Reviewer: ```suggestion
A file needs to be added for the date time formats.
```

> Follow the structure of other languages available

Reviewer: ```suggestion
Follow the structure of other languages available.
```

**nuxt.config.js (frontend/nuxt.config.js)**

> Two lines needs to be added here, one in the i18n locales

Reviewer: ```suggestion
Two lines needs to be added here, one in the i18n locales:
```

```
    { code: "[locale]", file: "[locale].json" },
```

> and one in the i18n dateTimeFormats :

Reviewer: ```suggestion
and one in the i18n `dateTimeFormats`:
```

```
    "[locale]": require("./lang/dateTimeFormats/[locale].json"),
```

**vuetify.options.js (frontend/vuetify.options.js)**

> Add a lang locale

Reviewer: ```suggestion
Add a lang locale:
```

```
    "[locale]": locale.[language],
```

**validators.py (mealie/schema/_mealie/validators.py)**

> Add a language for the schema validators

Reviewer: ```suggestion
Add a language for the schema validators:
```
